client/src/components/formHelpers/form.tsx

```diff
@@ -35,43 +36,58 @@ function validateFormValues(
   formValues: FormValues,
   options: FormOptions
 ): ValidatedValues {
-  const { isEditorLinkAllowed, isLocalLinkAllowed, types } = options;
+  const {
+    isEditorLinkAllowed,
+    isLocalLinkAllowed,
+    isSourceCodeLinkRequired,
+    types
+  } = options;
   const validatedValues: ValidatedValues = {
     values: {},
     errors: [],
     invalidValues: []
   };
-  const urlValues = Object.entries(formValues).reduce(
-    (result, [key, value]) => {
-      // NOTE: pathValidator is not used here, because it is only used as a
-      // suggestion - should not prevent form submission
-      const validators = [fCCValidator, httpValidator];
-      const isSolutionLink = key !== 'githubLink';
-      if (isSolutionLink && !isEditorLinkAllowed) {
-        validators.push(editorValidator);
-      }
-      if (!isLocalLinkAllowed) {
-        validators.push(localhostValidator);
-      }
 
-      const nullOrWarning = composeValidators(...validators)(value);
-      if (nullOrWarning) {
-        validatedValues.invalidValues.push(nullOrWarning);
-      }
-      if (value && types && types[key] === 'url') {
-        try {
-          value = normalizeUrl(value, normalizeOptions);
-        } catch (err: unknown) {
-          validatedValues.errors.push({
-            error: err as { message?: string },
-            value
-          });
-        }
+  const formFields = Object.entries(formValues);
+
+  if (isSourceCodeLinkRequired) {
+    const githubLink = formValues['githubLink'];
+    if (!githubLink) {
+      validatedValues.invalidValues.push(sourceCodeLinkExistsValidator(''));
+    }
+  }
```

Comment:
This is ready for review, but I was not happy with this section of code. So, suggestions would be helpful.

Reply:
Yeah, the form code as a whole is nasty.

What should happen is that the Form's `onSubmit` callback should get all the fields, but it doesn't. Not if you add and then delete a source code link. No idea why.

Given that context, I think this is fine, but we need a comment to explain because this makes no sense at first glance.